src/main/java/org/codeforamerica/messaging/config/SecurityConfiguration.java

```diff
@@ -28,4 +43,25 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .httpBasic(withDefaults());
         return http.build();
     }
+
+    private AuthorizationManager<RequestAuthorizationContext> checkForAllowedIp() {
+        return (authentication, context) -> {
+            HttpServletRequest request = context.getRequest();
+            boolean ipAddressAllowed = Arrays.stream(allowedIpAddresses.split(","))
+                    .anyMatch(allowedIpAddress -> {
+                        IpAddressMatcher ipAddressMatcher = new IpAddressMatcher(allowedIpAddress);
+                        String xForwardedFor = request.getHeader("X-Forwarded-For");
+                        boolean matchesXForwardedFor = false;
+                        if (xForwardedFor != null) {
+                            matchesXForwardedFor = Arrays.stream(xForwardedFor.split(", "))
+                                    .anyMatch(xForwardedForIp -> {
+                                        return ipAddressMatcher.matches(xForwardedForIp);
```

Comment:
Since you took out the log statements you can probably collapse this down to a method reference? not important